result.go

```diff
@@ -112,7 +112,7 @@ func Test(t *testing.T, description string, testFunc func()) {
 func (r *result) setLabels(t *testing.T) {
 	wsd := os.Getenv(wsPathEnvKey)
 
-	_, testFile, _, _ := runtime.Caller(2)
+	_, testFile, _, _ := runtime.Caller(3)
```

Comment:
Caller depth varies depending on which function is called. `allure.Test` has one more layer compared to `allure.TestWithParameters` and as a result I have the following situation:
![Screen Shot 2019-11-06 at 1 55 09 PM](https://user-images.githubusercontent.com/1215522/68328343-175c1e80-009d-11ea-9060-d58b0d3f084e.png)

We need a more dynamic approach for this problem. Maybe iterating through callers?

Reply:
Check my commit to this PR, it should do the trick.